src/server/getPatronData.ts

```diff
@@ -0,0 +1,105 @@
+import { isEmpty } from "underscore"
+
+import nyplApiClient from "./nyplApiClient"
+
+type NoticePreference = "None" | "Email" | "Telephone"
+interface PatronProps {
+  isTokenValid: boolean
+  userId: string
+}
+interface PatronObject {
+  id: string
+  names: string[]
+  barcodes: string[]
+  emails: string[]
+  loggedIn: boolean
+  moneyOwed?: number
+  homeLibraryCode?: string
+  patronType?: string
+  expirationDate?: string
+  phones?: string[]
+  noticePreference?: NoticePreference
+}
+
+/**
+ * extractNoticePreference(fixedFields)
+ * Returns 'None', 'Email', or 'Telephone'.
+ * @param {string} fixedFields - Object coming from patron object
+ *  `fixedFields` property
+ */
+export function extractNoticePreference(fixedFields = {}): NoticePreference {
+  if (isEmpty(fixedFields)) return "None"
+
+  const noticePreferenceMapping = {
+    z: "Email",
+    p: "Telephone",
+    "-": "None",
+  }
+  const noticePreferenceField = fixedFields["268"]
+
+  if (!noticePreferenceField?.value) return "None"
+  return noticePreferenceMapping[noticePreferenceField.value] || "None"
+}
+
+export async function getPatronData({
+  isTokenValid = false,
+  userId,
+}: PatronProps) {
+  let patron: PatronObject = {
+    id: "",
+    names: [],
+    barcodes: [],
+    emails: [],
+    loggedIn: false,
+  }
+
+  if (isTokenValid && userId) {
+    await nyplApiClient().then((client) => {
+      client
+        .get(`/patrons/${userId}`, { cache: false })
+        .then((response) => {
+          if (!isEmpty(response)) {
+            const {
+              id,
+              names,
+              emails,
+              moneyOwed,
+              homeLibraryCode,
+              patronType,
+              expirationDate,
+              phones,
+            } = response.data
+            const barcodes = response.data.barCodes
+            const noticePreference = extractNoticePreference(
+              response.data.fixedFields
+            )
+            // Data exists for the Patron
+            patron = {
+              id,
+              names,
+              barcodes,
+              emails,
+              loggedIn: true,
+              moneyOwed,
+              homeLibraryCode,
+              patronType,
+              expirationDate,
+              phones,
+              noticePreference,
+            }
```

Comment:
I think `patron` needs to be returned?

Reply:
I'm not sure that we actually want to return patron data here. The plan for My Account is to do fetches from Sierra instead of platform.

Reply:
Hmm true. Then maybe all this PR needs to do is the set whether the parsed and jwt verified token is valid since the patron id is there and this entire function is not needed.